pkg/controller/api/radapp.io/v1alpha3/deployment_webhook.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/radius-project/radius/pkg/ucp/ucplog"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+// SetupWebhookWithManager sets up a webhook for the built in Deployment resource with the given controller manager.
+// It creates a new webhook managed by the controller manager, registers the built in Deployment resource with the webhook,
+// sets the validator for the webhook to the Deployment instance, and completes the webhook setup.
+func (d *BuiltInDeployment) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(&appsv1.Deployment{}).
+		WithDefaulter(d).
+		Complete()
+}
+
+// +kubebuilder:webhook:path=/mutate-radapp-io-v1alpha3-builtindeployment,mutating=true,failurePolicy=fail,sideEffects=None,groups=radapp.io,resources=deployments,verbs=create;update,versions=v1alpha3,name=deployment-webhook.radapp.io,sideEffects=None,admissionReviewVersions=v1
+
+// BuiltInDeployment is a type that implements the mutating webhook function for the type Deployment.
+type BuiltInDeployment struct{}
```

Comment:
What I was expecting here is for us to implement a custom webhook that handles the built-in `apps/v1 Deployment` Kubernetes type. 

https://kubernetes.io/docs/concepts/workloads/controllers/deployment/

I'm good with however you want to sequence the changes. If this is going to be addressed in the next step, that's fine.

Reply:
If we see ln 34: https://github.com/radius-project/radius/blob/0077cb8ca4a39ea8bca0d182bfe3f19238a05626/pkg/controller/api/radapp.io/v1alpha3/deployment_webhook.go#L34 , this is where we specify that the webhook is for the built-in deployment type. I need to specify a different type on ln 42 as I cannot define new method (Default) on the built-in type